storage: add (disabled) post-split assertion

tbg opened the pull request:
I have written variations on these assertions a handful of times now, and
prefer not to have to do it again in the future.

Release note: None

tbg committed:
storage: handle error early

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21722)
<!-- Reviewable:end -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21722#-:-L3Zoq1uTw8CSR72KCEn:bnfp4nl)*
<!-- Sent from Reviewable.io -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21722#-:-L3ZpBicrXvICyuPuZnH:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: add (disabled) post-split assertion

I have written variations on these assertions a handful of times now, and
prefer not to have to do it again in the future.

Release note: None
